File: solutions_5751500831719424_0/C++/Taehyun/1B_A.cpp
#include<stdio.h>
#include<string.h>
#include<math.h>
#include<algorithm>
struct LIST
{
	int L;
	char A[101];
}list[101];
int N,M;
int X[101];
int Y[101];
int main()
{
	freopen("input.txt","r",stdin);
	freopen("output.txt","w",stdout);
	int t,T;
	char c;
	int flag;
	int Ans;
	scanf("%d",&T);
	for(t=1;t<=T;t++)
	{
		Ans=0;
		scanf("%d",&N);
		int i,j;
		for(i=1;i<=N;i++)
		{
			scanf("%s",&list[i].A[1]);
			list[i].L=strlen(&list[i].A[1]);
			X[i]=1;
		}
		flag=0;
		while(X[1]<=list[1].L)
		{
			c=list[1].A[X[1]];
			for(i=1;i<=N;i++)
			{
				Y[i]=0;
				for(;X[i]<=list[i].L;X[i]++)
				{
					if(list[i].A[X[i]]!=c) break;
					Y[i]++;
				}
				if(Y[i]==0) flag=1;
			}
			std::sort(Y+1,Y+1+N);
			for(i=1;i<=N;i++) Ans+=abs(Y[i]-Y[N/2+1]);
		}
		for(i=1;i<=N;i++) if(X[i]!=list[i].L+1) flag=1;
		printf("Case #%d: ",t);
		if(flag) printf("Fegla Won\n");
		else printf("%d\n",Ans);
	}
}

File: solutions_5751500831719424_1/Python/csurfer/1B_A.py
from io import open

def process(inp):
  n = int(inp.readline())
  stArr = [list(inp.readline())[:-1] for i in xrange(n)]
  uniqueArr = []
  for st in stArr:
    arr = []
    for ch in st:
      if len(arr) == 0:
        arr.append(ch)
      elif ch != arr[-1]:
        arr.append(ch)
    uniqueArr.append(arr)
  
  # Test Immpossibility
  for i in xrange(1,len(uniqueArr)):
    if uniqueArr[i] != uniqueArr[i-1]:
      return "Fegla Won"
  
  # Get minimum
  stCnt = []
  for i in xrange(len(uniqueArr)):
    #print stArr[i]
    tempArr = []
    while len(uniqueArr[i])>0:
      ch = uniqueArr[i][0]
      uniqueArr[i] = uniqueArr[i][1:]
      cnt = 0
      while len(stArr[i])>0:
        if ch == stArr[i][0]:
          cnt += 1
          stArr[i] = stArr[i][1:]
        else:
          break
      tempArr.append(cnt)
    stCnt.append(tempArr)
    #print tempArr

  # Calculate Min
  flag = True
  for i in xrange(1,len(stCnt)):
    if stCnt[i] != stCnt[i-1]:
      flag = False
  if flag:
    return 0

  action = 0
  for i in xrange(len(stCnt[0])):
    calArr = [el[i] for el in stCnt]
    calArr = sorted(calArr)
    if len(calArr)%2 == 0:
      # Even case
      action1 = 0
      action2 = 0
      median = calArr[(len(calArr)-1)//2]
      for num in calArr:
        action1 += abs(median-num)
      median = calArr[((len(calArr)-1)//2)+1]
      for num in calArr:
        action2 += abs(median-num)
      action += min(action1, action2)
    else:
      # Odd case
      median = calArr[len(calArr)//2]
      for num in calArr:
        action += abs(median-num)
  return action

def output(strOut, out):
  print strOut
  out.write(u"{}\n".format(strOut))

def main():
  inp = open("input.txt")
  out = open("output.txt", "wb+")
  tC = int(inp.readline())
  tc = tC
  while tc > 0:
    tc -= 1
    ans = process(inp)
    output(u"Case #{}: {}".format(tC-tc, ans), out)
  inp.close()
  out.close()

if __name__ == "__main__":
  main()
